mm2src/mm2_core/src/mm_ctx.rs

```diff
@@ -568,6 +572,15 @@ impl MmArc {
             .map_err(|e| MmMetricsError::Internal(e.to_string()))?;
         prometheus::spawn_prometheus_exporter(self.metrics.weak(), address, shutdown_detector, credentials)
     }
+
+    pub fn get_scripthash_notification_handlers(&self) -> ScripthashNotificationHandlers {
+        if self.event_stream_configuration.is_some() {
+            let (sender, receiver): (AsyncSender<()>, AsyncReceiver<()>) = futures::channel::mpsc::channel(1);
```

Comment:
Setting capacity to `1` might be not enough depending on what changes will be made according to my other comments.